components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/AnalyticsUtils.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.synapse.rest.RESTConstants;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.usage.publisher.dto.enums.EVENT_TYPE;
+import org.wso2.carbon.apimgt.usage.publisher.dto.enums.FAULT_EVENT_TYPE;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityUtils;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import java.util.Map;
+
+/**
+ * Implementation of the util methods
+ */
+public class AnalyticsUtils {
+    private static final String ANALYTICS_REQUEST_TYPE = "analytics_request_type";
+    private static final String THROTTLED_OUT_REQUEST = "throttled_out_request";
+
+    public static void markedAsThrottled(MessageContext messageContext) {
+        messageContext.setProperty(ANALYTICS_REQUEST_TYPE, THROTTLED_OUT_REQUEST);
+    }
+
+    public static boolean isRequestThrottledOut(MessageContext messageContext) {
+        if (messageContext.getPropertyKeySet().contains(ANALYTICS_REQUEST_TYPE)) {
+            String type = (String) messageContext.getProperty(ANALYTICS_REQUEST_TYPE);
+            return THROTTLED_OUT_REQUEST.equals(type);
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean isAuthFaultRequest(int errorCode) {
+        return errorCode >= 900900 && errorCode < 901000;
+    }
+
+    public static boolean isThrottledFaultRequest(int errorCode) {
+        return errorCode >= 900800 && errorCode < 900900;
+    }
+
+    public static boolean isTargetFaultRequest(int errorCode) {
+        return errorCode >= 101500 && errorCode < 101600;
+    }
+
+    public static FAULT_EVENT_TYPE getFaultyType(MessageContext messageContext) {
+        int errorCode = (int) messageContext.getProperty(SynapseConstants.ERROR_CODE);
+        if (isAuthFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.AUTH;
+        } else if (isThrottledFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.THROTTLED;
+        } else if (isTargetFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.TARGET;
+        } else {
+            return FAULT_EVENT_TYPE.OTHER;
+        }
+    }
+
+    public static boolean isFaultRequest(MessageContext messageContext) {
+        return messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE);
+    }
+
+    public static boolean isSuccessRequest(MessageContext messageContext) {
+        return !messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE)
+                && APISecurityUtils.getAuthenticationContext(messageContext) != null;
+    }
+
+    public static boolean isResourceNotFound(MessageContext messageContext) {
+        if (messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE)) {
+            int errorCode = (int) messageContext.getProperty(SynapseConstants.ERROR_CODE);
+            return messageContext.getPropertyKeySet().contains(RESTConstants.PROCESSED_API) && errorCode == 404;
```

Comment:
Here too. At least it's better use some defined constants.

Reply:
Fixed